index.js

```diff
@@ -130,21 +135,25 @@ module.exports = {
       destDir: 'ember-data'
     });
 
-    var privateTree = babel.transpileTree(withPrivate, {
+    var privateTree = babel.transpileTree(this.debugTree(withPrivate, 'babel-private:input'), {
       babel: this.buildBabelOptions(),
       'ember-cli-babel': {
         compileModules: false
       }
     });
 
+    privateTree = this.debugTree(privateTree, 'babel-private:output');
+
     // use the default options
-    var publicTree = babel.transpileTree(withoutPrivate);
+    var publicTree = babel.transpileTree(this.debugTree(withoutPrivate, 'babel-public:input'));
+
+    publicTree = this.debugTree(publicTree, 'babel-public:output');
 
     privateTree = new Rollup(privateTree, {
       rollup: {
         entry: '-private/index.js',
         targets: [
-          { dest: '-private.js', format: 'amd', moduleId: 'ember-data/-private' }
+          { dest: 'ember-data/-private.js', format: babel.shouldCompileModules() ? 'amd' : 'es', moduleId: 'ember-data/-private' }
```

Comment:
In 2.12 and higher we simply concat regardless of source path, so this is fine.  In < 2.12, we need to be in the right place so that the transpiled AMD module has the right `moduleId`.

If you notice down below, the public modules are emitted into `modules/ember-data` already (in all cases). This should be fine for both sides (and the tests are passing).